frontend/reducers/song_index_reducer.js

```diff
@@ -7,13 +7,16 @@ import { RECEIVE_CURRENT_ALBUM,
 } from "../actions/album_actions";
 
 
-const songIndexReducer = (songsState = {}, action) => {
+const songIndexReducer = (songsState = [], action) => {
     Object.freeze(songsState);
     switch (action.type) {
         case RECEIVE_CURRENT_PLAYLIST:
             return action.songs;
         case RECEIVE_CURRENT_ARTIST:
-            return action.songs;
+            return {
+				allSongs: action.allSongs,
+				collabSongs: action.collabSongs,
+			};
```

Comment:
hmm seems like this reducer returns different types depending on the action? either an array of songs or an object, is this intentional? maybe you want `actions.allSongs.concat(actions.collabSongs)`

Reply:
Yes you understand correctly; depending on the action type (which corresponds with the page view), there are different payloads. Currently CollabSongs is actually a joins table between Artist and Song, and I needed it in its own key at the time of retrieval or else I wouldn't be able to distinguish it from the others. Does that make sense?

Reply:
Yeah makes sene